src/web/routes/application/middleware/handle-path-request/handle-path-request.js

```diff
@@ -18,13 +19,16 @@ const middleware = (config, pathsInSequence, step) => (req, res, next) => {
     return next()
   }
 
-  // Initialise nextAllowedStep if none exists in session
-  if (!req.session.nextAllowedStep) {
-    req.session.nextAllowedStep = pathsInSequence[0]
+  // Initialise nextAllowedPath if none exists in session
+  const firstPathInSequence = pathsInSequence[0]
+  let nextAllowedPath = stateMachine.dispatch(GET_NEXT_ALLOWED_PATH, req)
+
+  if (!nextAllowedPath) {
+    stateMachine.dispatch(SET_NEXT_ALLOWED_PATH, req, firstPathInSequence)
+    nextAllowedPath = firstPathInSequence
```

Comment:
Feels like there should be a cleaner way to do this without mutating the nextAllowedPath variable but can't see anything obvious without the code in front of me, have you tried a way without the mutation?

Also, `firstPathInSequence` is only used in the if block so lets move it into there.

Reply:
Yes, I agree - its possible but involves calling the state machine twice which seemed unnecessary.

Reply:
So the alternative would look something like:
```
const shouldInitialiseNextAllowedPath = (req) => isUndefined(stateMachine.dispatch(GET_NEXT_ALLOWED_PATH, req))

...

// Initialise nextAllowedPath if none exists in session
if (shouldInitialiseNextAllowedPath(req)) {
  stateMachine.dispatch(SET_NEXT_ALLOWED_PATH, req, pathsInSequence[0])
}

const nextAllowedPath = stateMachine.dispatch(GET_NEXT_ALLOWED_PATH, req)
const isPathAllowed = stateMachine.dispatch(IS_PATH_ALLOWED, req, pathsInSequence)
```
Another option would be to create a function that initialises the session outside of this middleware, which would negate the need for the `undefined` check.

Any thoughts on those options?

Reply:
Yes, I see what you mean, and now I see why the initial solution was written as it is. I wouldn't want to call the state machine twice, so I'm happy to leave it as it was initially just with the `firstPathInSequence` variable in the if block.